nixos/modules/services/monitoring/telegraf.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.telegraf;
+
+  configFile = pkgs.runCommand "config.toml" {
+    buildInputs = [ pkgs.remarshal ];
+  } ''
+    remarshal -if json -of toml \
+      < ${pkgs.writeText "config.json" (builtins.toJSON cfg.extraConfig)} \
+      > $out
+  '';
+in
+{
+
+  ###### interface
+
+  options = {
+
+    services.telegraf = {
+
+      enable = mkOption {
+        default = false;
+        description = "Whether to enable the telegraf server";
+        type = types.bool;
+      };
+
+      package = mkOption {
```

Comment:
But *why* is that useful in this case. What I'm getting at is that adding options just because you can should be avoided.